packages/storage/src/types/data.ts

```diff
@@ -0,0 +1,55 @@
+import { z } from "zod";
+
+function isBrowser() {
+  return typeof window !== "undefined";
+}
+
+const JsonLiteralSchema = z.union([
+  z.string(),
+  z.number(),
+  z.boolean(),
+  z.null(),
+]);
+
+type JsonLiteral = z.infer<typeof JsonLiteralSchema>;
+
+const FileOrBufferUnionSchema = isBrowser()
+  ? (z.instanceof(File) as z.ZodType<InstanceType<typeof File>>)
+  : (z.instanceof(Buffer) as z.ZodTypeAny); // TODO: fix, this is a hack to make browser happy for now
+
+export const FileOrBufferSchema = z.union([
+  FileOrBufferUnionSchema,
+  z.object({
+    data: z.union([z.instanceof(Buffer), z.string()]),
+    name: z.string(),
+  }),
+]);
+
+export type FileOrBuffer = File | Buffer | BufferOrStringWithName;
+
+export type BufferOrStringWithName = {
+  data: Buffer | string;
+  name: string;
+};
+
+export const FileOrBufferOrStringSchema = z.union([
+  FileOrBufferSchema,
+  z.string(),
+]);
+
+export type FileOrBufferOrString = FileOrBuffer | string;
+
+const JsonSchema: z.ZodType<Json> = z.lazy(() =>
+  z.union([
+    JsonLiteralSchema,
+    JsonObjectSchema,
+    FileOrBufferSchema,
+    z.array(JsonSchema),
+  ]),
+);
+
+export const JsonObjectSchema = z.record(z.string(), JsonSchema);
+
+export type Json = JsonLiteral | FileOrBuffer | JsonObject | Json[];
+
+export type JsonObject = { [key: string]: Json };
```

Comment:
I think i duplicated these in solana, can you check that we're importing from here in both ts SDK/solana?